helix-core/src/movement.rs

```diff
@@ -1445,6 +1452,100 @@ mod test {
         }
     }
 
+    #[test]
+    fn test_behaviour_when_moving_to_end_of_prev_long_words() {
+        let tests = [
+            (
+                "Basic backward motion from the middle of a word",
+                vec![(1, Range::new(3, 3), Range::new(4, 0))],
+            ),
+            ("Starting from after boundary retreats the anchor",
+                vec![(1, Range::new(0, 9), Range::new(8, 0))],
+            ),
```

Comment:
This was commented out in `test_behaviour_when_moving_to_end_of_next_long_words` but seems to be passing here so I added it back.